runtime/vm/UpcallThunkMem.cpp

```diff
@@ -87,66 +87,100 @@ subAllocateThunkFromHeap(J9UpcallMetaData *data)
 {
 	J9JavaVM * vm = data->vm;
 	UDATA thunkSize = data->thunkSize;
+	J9UpcallThunkHeapList *thunkHeapHead = vm->thunkHeapHead;
+	J9UpcallThunkHeapList *thunkHeapNode = NULL;
+	J9UpcallThunkHeapWrapper *thunkHeapWrapper = NULL;
 	J9Heap *thunkHeap = NULL;
 	void *subAllocThunkPtr = NULL;
+	bool succeeded = false;
 	PORT_ACCESS_FROM_JAVAVM(vm);
 
 	Trc_VM_subAllocateThunkFromHeap_Entry(thunkSize);
 
-	if (NULL == vm->thunkHeapWrapper) {
-		thunkHeap = allocateThunkHeap(data);
-		if (NULL == thunkHeap) {
+	if (NULL == thunkHeapHead) {
+retry:
+		thunkHeapHead = allocateThunkHeap(data);
+		if (NULL == thunkHeapHead) {
 			/* The thunk address is NULL if VM fails to allocate the thunk heap */
 			goto done;
 		}
-	} else {
-		thunkHeap = vm->thunkHeapWrapper->heap;
 	}
-
-#if defined(OSX) && defined(AARCH64)
-	pthread_jit_write_protect_np(0);
-#endif
-	subAllocThunkPtr = j9heap_allocate(thunkHeap, thunkSize);
-	if (NULL != subAllocThunkPtr) {
-		J9UpcallMetaDataEntry metaDataEntry = {0};
-		metaDataEntry.thunkAddrValue = (UDATA)(uintptr_t)subAllocThunkPtr;
-		metaDataEntry.upcallMetaData = data;
-
-		/* Store the address of the generated thunk plus the corresponding metadata as an entry to
-		 * a hashtable which will be used to release the memory automatically via freeUpcallStub
-		 * in OpenJDK when their native scope is terminated or VM exits.
-		 */
-		if (NULL == hashTableAdd(vm->thunkHeapWrapper->metaDataHashTable, &metaDataEntry)) {
-			j9heap_free(thunkHeap, subAllocThunkPtr);
-			subAllocThunkPtr = NULL;
-			Trc_VM_subAllocateThunkFromHeap_suballoc_thunk_add_hashtable_failed(thunkSize, thunkHeap);
+	thunkHeapNode = thunkHeapHead;
+
+	while (!succeeded) {
+		thunkHeapWrapper = thunkHeapNode->thunkHeapWrapper;
+		thunkHeap = thunkHeapWrapper->heap;
+
+	#if defined(OSX) && defined(AARCH64)
```

Comment:
As a rule, we do not indent preprocessor directives.

Reply:
Fixed everywhere in this PR.